src/pyflex/window_selector.py

```diff
@@ -505,6 +543,45 @@ def calculate_ttimes(self):
         self.ttimes = sorted(tts, key=lambda x: x["time"])
         logger.info("Calculated travel times.")
 
+    def reject_on_select_mode(self):
```

Comment:
This is really cool.

I think it could also be expanded to select windows for specific phase arrivals. TauP does already calculate the arrival times for various phases so this function could well filter windows that contain the chosen phases.